src/code/z_actor.c

```diff
@@ -2707,19 +2708,88 @@ void Actor_DrawLensActors(PlayState* play, s32 numInvisibleActors, Actor** invis
     CLOSE_DISPS(gfxCtx, "../z_actor.c", 6284);
 }
 
-s32 func_800314B0(PlayState* play, Actor* actor) {
-    return func_800314D4(play, actor, &actor->projectedPos, actor->projectedW);
+/**
+ * Checks if an actor should be culled or not, by seeing if it is contained within its own culling volume.
+ * For more details on the culling test, see `Actor_CullingVolumeTest`.
+ *
+ * "Culling" in this context refers to the removal of something for the sake of improving performance.
+ * For actors, being culled means that their Update and Draw processes are halted.
+ * While halted, an Actor's update state is frozen and it will not draw, making it invisible.
+ *
+ * Actors that are within the bounds of their culling volume may update and draw, while actors that are
+ * out of bounds of its culling volume may be excluded from updating and drawing until they are within bounds.
+ *
+ * It is possible for actors to opt out of update culling or draw culling.
+ * This is set per-actor with `ACTOR_FLAG_UPDATE_CULLING_DISABLED` and `ACTOR_FLAG_DRAW_CULLING_DISABLED`.
+ *
+ * Note: Even if either `ACTOR_FLAG_UPDATE_CULLING_DISABLED` or `ACTOR_FLAG_DRAW_CULLING_DISABLED` are set, the actor
+ * will still undergo the culling test and set `ACTOR_FLAG_INSIDE_CULLING_VOLUME` accordingly.
+ * So, `ACTOR_FLAG_INSIDE_CULLING_VOLUME` cannot be used on it own to determine if an actor is actually culled.
+ * It simply says whether or not they are physically located within the bounds of the culling volume.
+ */
+s32 Actor_CullingCheck(PlayState* play, Actor* actor) {
+    return Actor_CullingVolumeTest(play, actor, &actor->projectedPos, actor->projectedW);
 }
 
-s32 func_800314D4(PlayState* play, Actor* actor, Vec3f* arg2, f32 arg3) {
-    f32 var;
+/**
+ * Tests if an actor is currently within the bounds of its own culling volume.
+ *
+ * The culling volume is a 3D shape composed of a frustum with a box attached to the end of it. The frustum sits at the
+ * camera's position and projects forward, encompassing the player's current view; the box extrudes behind the camera,
+ * allowing actors in the immediate vicinity behind and to the sides of the camera to be detected.
+ *
+ * This function returns true if the actor is within bounds, false if not.
+ * The comparison is done in projective space against the actor's projected position as the viewing frustum
+ * in world space transforms to a box in projective space, making the calculation easy.
```

Comment:
IMO
"projective space" -> "projected space"
"projective properties" -> "projected properties"

to keep terminology consistent with z64actor.h. In my mind "projective space" refers to the mathematical object (like "Euclidean space"), while "projected space" I guess means the specific coordinate system (like "model space"). I'm more familiar with the term "clip space" for the latter, but I guess I can get used to "projected space"

Reply:
@Thar0 wrote this specifically and will defer (whenever he finds the time)

Reply:
I wouldn't mind clip space to make it clear it's referring to a specific coordinate system, but I also think it doesn't matter since there is exactly one projective coordinate system used by the game unlike the various affine coordinate systems in use.

Reply:
"projected properties" doesn't really make sense as a term, I think it's better to call the properties "projective" still even if we're talking about a specific projective coordinate system

Reply:
We can avoid that term entirely by changing

> Every actor can set properties for their own culling volume, changing its dimensions to suit the needs of it and its environment. Note that the following properties are projective properties as they are compared with the actor's position after perspective projection is applied. The units are therefore not directly comparable to world units.

to something like

> Every actor can set properties for their own culling volume, changing its dimensions to suit the needs of it and its environment. These properties are in units of projected space (i.e. compared to the actor's position after perspective projection is applied) are therefore not directly comparable to world units.

I already think we've spent too much time on this though, really the only thing I care about is that "projected space"/"projective space" matches between z64actor.h and here

Reply:
Im fine with the suggestion, tharo?

Reply:
Sure that looks good